node/network/gossip-support/src/metrics.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use polkadot_node_subsystem_util::{
+	metrics,
+	metrics::{
+		prometheus,
+		prometheus::{Gauge, PrometheusError, Registry, U64},
+	},
+};
+
+/// Dispute Distribution metrics.
+#[derive(Clone, Default)]
+pub struct Metrics(Option<MetricsInner>);
+
+#[derive(Clone)]
+struct MetricsInner {
+	/// Tracks authority status.
+	is_authority: Gauge<U64>,
+}
+
+impl Metrics {
+	/// Dummy constructor for testing.
+	#[cfg(test)]
+	pub fn new_dummy() -> Self {
+		Self(None)
+	}
+
+	/// Set the authority flag.
+	pub fn on_is_authority(&self) {
```

Comment:
I think authority is too ambiguous, what we are checking is "parachain validator".
Authority also includes block producers outside of parachain validators.

Reply:
Yes, but "parachain validator" doesn't sound accurate to me, since the validator(authority) in our context is a block producer which includes validating the parachain candidates.

Reply:
Glossary[^1] actually says that "Parachain Validators" are the backing groups 
> A subset of validators assigned during a period of time to back candidates for a specific parachain

However, note that of all the relay chain validators (1000 on Kusama), we select maxValidators (200 on Kusama) and only these do approval-checking of parachains. The relay chain block produces may or may not be in this set at any given time. We may eventually set maxValidators to None, but I wanted to point out `authorities` is the superset of what we're checking.

[^1]: https://w3f.github.io/parachain-implementers-guide/glossary.html

Reply:
Thanks for posting this. That means I have to change the code so that we're checking the superset or update the metric definition and naming so it makes sense.

Reply:
I don't wan to bikeshed even more, but I find that `role_authority` irritating as in [other places](https://github.com/paritytech/polkadot/blob/22e5c04008dd85ddd36c4af74e3bdcdb20bb0645/node/service/src/lib.rs#L726) "authority role" just means the node has been started with `--validator`, regardless on whether the node has actual keys and in fact is a validator.

:grimacing:

Reply:
In my understanding we have been quite consistent in the code base to talk about authorities, meaning any potential block producer (has `AuthorityDiscoveryKey`) and (parachain-)validators having a key in `SessionInfo::validators`.

Reply:
IMO this metric should not actually be tracking the cli arguments. For this to fully make sense I'll have to include 2 metrics - one will say if the node is currently a relay chain validator while the other will say if the node is in the approval checking group. I'll fix the naming along the way to make it consistent with what I said above.

Reply:
done, please take another look